File: java-sdk/lib/src/main/java/models/shared/FileVersionMini.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * FileVersionMini
 * The bare basic representation of a file version, the minimal
 * amount of fields returned when using the `fields` query
 * parameter.
**/
public class FileVersionMini {
    @JsonProperty("id")
    public String id;
    public FileVersionMini withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sha1")
    public String sha1;
    public FileVersionMini withSha1(String sha1) {
        this.sha1 = sha1;
        return this;
    }
    @JsonProperty("type")
    public FileVersionMiniTypeEnum type;
    public FileVersionMini withType(FileVersionMiniTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: python-sdk/src/boxsdk/models/shared/fileversion_mini.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class FileVersionMiniTypeEnum(str, Enum):
    FILE_VERSION = "file_version"


@dataclass_json
@dataclasses.dataclass
class FileVersionMini:
    r"""FileVersionMini
    The bare basic representation of a file version, the minimal
    amount of fields returned when using the `fields` query
    parameter.
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: FileVersionMiniTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    sha1: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sha1') }})
    
